test/functional/rpc_psbt.py

```diff
@@ -660,6 +663,55 @@ def test_psbt_input_keys(psbt_input, keys):
                 break
         assert shuffled
 
+        self.log.info("Test descriptorprocesspsbt updates and signs a psbt with descriptors")
+        key_info = get_generate_key()
+        key1 = key_info[0]
+        wpkh_addr = key_info[5]
+        descriptor1 = descsum_create(f"wpkh({key1})")
+
+        a_key_info = get_generate_key()
+        b_key_info = get_generate_key()
+        c_key_info = get_generate_key()
+        a_key = a_key_info[0]
+        a_pubkey = a_key_info[1]
+        b_key = b_key_info[0]
+        b_pubkey = b_key_info[1]
+        c_pubkey = c_key_info[1]
+
+        descriptor_a = descsum_create(f"wsh(multi(2,{a_key},{b_pubkey},{c_pubkey}))")
```

Comment:
Can you leave comments on each step to make it clear what's happening? It's been a few minutes of staring and I'm not quite sure what's exactly being tested here, making it difficult to update in the future.

My basic test flow would probably follow the lines of:

1) pick a node that doesn't have a wallet
2) make a single wpkh utxo
3) make single input psbt with that utxo
4) make sure it signs it when expected, and finalizes when expected (e.g., give it a descriptor unrelated to the utxo, partial_signature lying around when expected, vary sighash modes)
5) make sure extracted tx submits to mempool ok

imo we don't need to re-test all of the descriptor logic that's internal and hopefully already tested; we just want to check that the API here makes sense and pipes through. The 2-of-3 multisig, multi-input stuff just seems like a distraction.

Reply:
I've rewritten the test to reflect this. I've also added more comments.

Reply:
great, thanks!